src/main/java/org/example/Game.java

```diff
@@ -1,181 +1,262 @@
+      System.out.println(result);
+      player.symbol = this.sc.nextLine().toUpperCase().charAt(0);
+      if (player.symbol == 'Q') {
+        println("Are u leaving me in th middle ?? ohk then \uD83D\uDC4B \uD83D\uDC4B");
+        exit(0);
+      }
+
+    } while (!(player.symbol == X_SYMBOL || player.symbol == O_SYMBOL)
+        && println("Sry, Cannot enter anything except X_SYMBOL or O_SYMBOL \uD83E\uDD25 !"));
+  }
 
-    void initialStartup() {
+  protected void initialStartup() {
 
-        manual();
-        boardobj.initializeBoard();
-        boardobj.showBoard();
-        player.setName();
-        player.setSymbol();
-        cyborg.setSymbol( (player.getSymbol() == 'X')?'O':'X');
-        startPlay();
+    manual();
+    setPlayerName();
+    postStartup();
   }
 
-    private  void startPlay()  {
+  private void postStartup() {
+    setPlayerSymbol();
+    cyborg.setSymbol((player.getSymbol() == X_SYMBOL) ? O_SYMBOL : X_SYMBOL);
+    boardobj.initializeBoard();
+    boardobj.showBoard();
+    startPlay();
+  }
 
-        if (player.getSymbol() == 'X') {
-            while (true) {
-                playerPlay();
-                computerPlay();
-            }
+  private void startPlay() {
 
-        } else {
-            while (true) {
-                computerPlay();
-                playerPlay();
-            }
-        }
+    if (player.getSymbol() == X_SYMBOL) {
+      while (true) {
+        playerPlay();
+        computerPlay();
+      }
 
+    } else {
+      while (true) {
+        computerPlay();
+        playerPlay();
+      }
     }
+  }
 
+  private int getPositionFromUser() {
+    int position = -1;
 
+    do {
+      try {
 
-    private  void computerPlay() {
+        String result =
+            String.format(
+                "%s enter empty location for the next move from 1 to 9 : ", player.getName());
+        System.out.println(result);
+        position = this.sc.nextInt();
+        if (!(position >= 1 && position <= 9)) {
+          position = -1;
+          throw new OutOfRange(" \uD83D\uDE4F Plzz enter position between 0 to 9 only !");
+        } else if (!boardobj.getEmptyPositions().contains(position)) {
+          position = -1;
+          throw new OutOfPosition(" \uD83D\uDE4F Plzz enter an empty position !");
+        }
 
-        Iterator<Integer> i = boardobj.getIntList().iterator();
-        int position = boardobj.getIntList().get(0);
-        boardobj.getIntList().remove(0);
-        int row = (position - 1) / 3;
-        int col = position - (row * 3) - 1;
-        boardobj.getBoard()[row][col] = cyborg.getSymbol();
-        System.out.println("~~~~~~~~~~~~~~~~~~~~~~~~~~~  Cyborg's Move  ~~~~~~~~~~~~~~~~~~~~~~~~~~");
-        boardobj.showBoard();
-        isWinningMoveOrTie(cyborg.getSymbol());
-    }
+      } catch (InputMismatchException e) {
+        println(" \uD83D\uDE4F Plzz Enter only numbers  from 1 to 9 !! ");
+
```

Comment:
Again over here in print state you have made it moves from 1 to 9. For 4X4 or 5X5 you need to change you print statement everytime. So write print state in the way that it can also be sed for nXn.

Reply:
done